app/src/main/java/org/gdg/frisbee/android/api/Callback.java

```diff
@@ -0,0 +1,42 @@
+package org.gdg.frisbee.android.api;
+
+import android.support.annotation.StringRes;
+
+import org.gdg.frisbee.android.R;
+
+import java.io.IOException;
+
+import retrofit.Response;
+import retrofit.Retrofit;
+import timber.log.Timber;
+
+public abstract class Callback<T> implements retrofit.Callback<T> {
+
+    @Override
+    public final void onResponse(Response<T> response, Retrofit retrofit) {
+        if (response.isSuccess()) {
+            success(response.body());
+        } else {
+            try {
+                final Exception e = new Exception(response.errorBody().string());
+                Timber.e(e, "Network Error!");
+                failure(e, R.string.server_error);
+            } catch (IOException e) {
+                Timber.e(e, "Network Error!");
+                failure(e, R.string.server_error);
+            }
+        }
+    }
+
+    @Override
+    public final void onFailure(Throwable t) {
+        Timber.d(t, "Network Failure!");
```

Comment:
This means that there is problem with the connection. This will just send a debug log and show network error message.